src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -185,26 +172,35 @@ public InternalEngine(EngineConfig engineConfig, boolean skipInitialTranslogReco
         logger.trace("created new InternalEngine");
     }
 
+    private Translog openTranslog(EngineConfig engineConfig, IndexWriter writer, boolean createNew) throws IOException {
+        final Translog.TranslogGeneration generation = loadTranslogIdFromCommit(writer);
+        Translog translog;
+        TranslogConfig translogConfig = engineConfig.getTranslogConfig();
+        if (createNew) {
+            translog = new Translog(translogConfig);
+        } else {
+            translogConfig.setTranslogGeneration(generation);
+            if (generation != null && generation.translogUUID == null) {
+                // only upgrade on pre-2.0 indices...
+                Translog.upgradeLegacyTranslog(logger, translogConfig);
+            }
+            translog = new Translog(translogConfig);
```

Comment:
Can we pull out this line to outside the if/else since we do the same thing in both places?